src/main/java/mesosphere/dcos/client/DCOSAuthTokenHeaderInterceptor.java

```diff
@@ -28,4 +30,12 @@ public void apply(RequestTemplate template) {
 
         template.header("Authorization", "token=" + dcosAuthToken.getToken());
     }
+
+    public boolean hasToken() {
```

Comment:
These methods make me consider the thread safety of RequestInterceptor. I didn't really even think about it when we first created this interceptor, but it does hold state internally (the auth token). Since only a single instance of this is created and applied to all requests, it's likely that this will get called from possible threads.

I think before it wouldn't have been too big a deal, as the only consequence that I can see would have been calling authenticate twice (i.e. thread 1 checks if authToken==null, thread 2 interrupts and checks if authToken==null, and they both see this state as true so continue on and each call authenticate - the token would just be set to whatever set it last).

Now we may have the case where another thread is clearing out a token after another thread has refreshed it already, causing a request to be made with no token.

If we were to fix this with too broad synchronization, then it could negatively impact performance... probably need to do something like a double-check lock. 

Think we probably don't need to worry about this too much though, as it seems pretty unlikely, and even if it occurs it should correct itself quickly.